### ISSUE TITLE
[Client] Crash in TLSSensingCallback::getReadBuffer

### ISSUE THREAD

**Issue by zhengxiaochuan-3** [facebookincubator/LogDevice#128]
```
W1013 22:05:03.866642      55 [ld:srv:WG4] RebuildingSupervisor.cpp:679] adjustRebuildingThrottle() Entering throttling mode: #draining=0, #rebuilding=0, #triggers=2, #adjusted cluster size=4, threshold=35%, actual=50%
E1013 22:07:35.767406      11 [not-set] ConnectionListener.cpp:71] handlerReady() Error peeking header message from client socket
C1013 22:07:36.035210      12 [not-set] CommandListener.cpp:53] getReadBuffer() There is no 1 byte in the socket for TLS detection.
handle_fatal_signal(): Caught coredump signal 11
E1013 22:07:36.161312      13 [not-set] ConnectionListener.cpp:71] handlerReady() Error peeking header message from client socket
    @ 00005585b36f9d68 facebook::logdevice::handle_fatal_signal(int)
                       -> /data/LogDevice/logdevice/external/folly/folly/experimental/symbolizer/Elf.cpp
    @ 00005585b2d5a053 coredump_signal_handler(int)
                       -> /data/LogDevice/logdevice/external/folly/folly/experimental/symbolizer/Elf.cpp
    @ 00007ff18a69788f (unknown)
    @ 00005585b2e50a66 std::__uniq_ptr_impl<folly::AsyncSocket, folly::DelayedDestruction::Destructor>::_M_ptr() const
                       /usr/include/c++/7/bits/unique_ptr.h:147
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSSLSocket.cpp
    @ 00005585b2e4f859 std::unique_ptr<folly::AsyncSocket, folly::DelayedDestruction::Destructor>::get() const
                       /usr/include/c++/7/bits/unique_ptr.h:337
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSSLSocket.cpp
    @ 00005585b2e4f4fd std::unique_ptr<folly::AsyncSocket, folly::DelayedDestruction::Destructor>::release()
                       /usr/include/c++/7/bits/unique_ptr.h:359
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSSLSocket.cpp
    @ 00005585b2e4e0b7 std::unique_ptr<folly::AsyncSocket, folly::DelayedDestruction::Destructor>::unique_ptr(std::unique_ptr<folly::AsyncSocket, folly::DelayedDestruction::Destructor>&&)
                       /usr/include/c++/7/bits/unique_ptr.h:239
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSSLSocket.cpp
    @ 00005585b2e4981f facebook::logdevice::AdminCommandConnection::TLSSensingCallback::getReadBuffer(void**, unsigned long*)
                       -> /data/LogDevice/logdevice/external/folly/folly/experimental/symbolizer/Elf.cpp
    @ 00005585b30dcb3c folly::AsyncSocket::prepareReadBuffer(void**, unsigned long*)
                       /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSocket.cpp:1874
    @ 00005585b30dd5f9 folly::AsyncSocket::handleRead()
                       /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSocket.cpp:1989
    @ 00005585b30dc2da folly::AsyncSocket::ioReady(unsigned short)
                       /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSocket.cpp:1811
    @ 00005585b30e3ce8 folly::AsyncSocket::IoHandler::handlerReady(unsigned short)
                       /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSocket.h:1051
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/AsyncSocket.cpp
    @ 00005585b31057f6 folly::EventHandler::libeventCallback(int, short, void*)
                       /data/LogDevice/logdevice/external/folly/folly/io/async/EventHandler.cpp:161
    @ 00007ff1898ae8f7 (unknown)
    @ 00007ff1898af33e event_base_loop
    @ 00005585b30f4c05 folly::EventBase::loopBody(int, bool)
                       /data/LogDevice/logdevice/external/folly/folly/io/async/EventBase.cpp:335
    @ 00005585b30f4711 folly::EventBase::loop()
                       /data/LogDevice/logdevice/external/folly/folly/io/async/EventBase.cpp:261
    @ 00005585b30f5b89 folly::EventBase::loopForever()
                       /data/LogDevice/logdevice/external/folly/folly/io/async/EventBase.cpp:474
    @ 00005585b3131585 folly::run(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&)
                       /data/LogDevice/logdevice/external/folly/folly/io/async/ScopedEventBaseThread.cpp:40
    @ 00005585b3132903 void std::__invoke_impl<void, void (*)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> >(std::__invoke_other, void (*&&)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*&&, folly::EventBase*&&, folly::Baton<true, std::atomic>*&&, folly::Range<char const*>&&)
                       /usr/include/c++/7/bits/invoke.h:60
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/ScopedEventBaseThread.cpp
    @ 00005585b3132370 std::__invoke_result<void (*)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> >::type std::__invoke<void (*)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> >(void (*&&)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*&&, folly::EventBase*&&, folly::Baton<true, std::atomic>*&&, folly::Range<char const*>&&)
                       /usr/include/c++/7/bits/invoke.h:95
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/ScopedEventBaseThread.cpp
    @ 00005585b3132f82 decltype (__invoke((_S_declval<0ul>)(), (_S_declval<1ul>)(), (_S_declval<2ul>)(), (_S_declval<3ul>)(), (_S_declval<4ul>)())) std::thread::_Invoker<std::tuple<void (*)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> > >::_M_invoke<0ul, 1ul, 2ul, 3ul, 4ul>(std::_Index_tuple<0ul, 1ul, 2ul, 3ul, 4ul>)
                       /usr/include/c++/7/thread:234
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/ScopedEventBaseThread.cpp
    @ 00005585b3132ee4 std::thread::_Invoker<std::tuple<void (*)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> > >::operator()()
                       /usr/include/c++/7/thread:243
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/ScopedEventBaseThread.cpp
    @ 00005585b3132ec3 std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> const&), folly::EventBaseManager*, folly::EventBase*, folly::Baton<true, std::atomic>*, folly::Range<char const*> > > >::_M_run()
                       /usr/include/c++/7/thread:186
                       -> /data/LogDevice/logdevice/external/folly/folly/io/async/ScopedEventBaseThread.cpp
    @ 00007ff1872a966e (unknown)
    @ 00007ff18a68c6da start_thread
    @ 00007ff18696688e clone
handle_fatal_signal(): processed segments: 108
handle_fatal_signal(): unmapped pages - unmap calls: 22 - 8
```

**Comment by MohamedBassem** [comment]
Hi @zhengxiaochuan-3 thanks for reporting the crash. We will try to repro and send a fix soon!
In the meanwhile, do you have anything that can help us repro? Or does it consistently happen?

**Comment by zhengxiaochuan-3** [comment]
This crash has only appeared once. I just create a new cluster ,no special operations. The timezone on logdevice containers and my client containers are different.

**Comment by mcrnic** [comment]
Hi @zhengxiaochuan-3, thanks for reporting. Fix will be in today.

**Comment by MohamedBassem** [comment]
@mcrnic landed the fix in 3ee7ca0bfb8a0a04c89a575f14b7fa64e848669e.
